dxp/sennajs_integration.md

```diff
@@ -0,0 +1,117 @@
+# Senna.js integration in DXP
+
+This article describes how [Senna.js](https://github.com/liferay/senna.js) is integrated in DXP and how it works under the hood.
+
+### Introduction
+
+The infrastructure around Senna.js is located in the [`frontend-js-spa-web`](https://github.com/liferay/liferay-portal/tree/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web) module.
+
+In this module, a Senna.js [`App`](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/app/App.js) instance is created, with two routes which are handled by a specific [`Screen`](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/screen/Screen.js) instance.
+
+-   The [`ActionURLScreen`](https://github.com/liferay/liferay-portal/blob/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/ActionURLScreen.es.js) class is used for all requests made to "action" URLs, which are used to post forms. It makes sure that the http method used is `POST` and checks that the `p_p_lifecycle` URL parameter has the correct value when submitting a form.
+
+-   The [`RenderURLScreen`](https://github.com/liferay/liferay-portal/blob/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/RenderURLScreen.es.js) class is used for all "render" URLs, which return content, and is set to cacheable.
+
+Both of these classes inherit from the [`EventScreen`](https://github.com/liferay/liferay-portal/blob/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/screen/EventScreen.es.js) class which adds specific behaviour on top of Senna.js's [`Screen`](https://github.com/liferay/senna.js/blob/059022b737de0f0108b7ef3f06f14d7ade3baa94/src/screen/Screen.js) class such as dispatching events during the screen's lifecycle (loading, activated, disposed) or managing the cache.
+
+The Senna.js `App` instance is exposed through the global variable [`Liferay.SPA.app`](https://github.com/liferay/liferay-portal/blob/2d4120b7fdb8cf03e7505ebdaacee3bdceec3391/modules/apps/frontend-js/frontend-js-spa-web/src/main/resources/META-INF/resources/liferay/init.es.js#L129) although that might change in the future
```

Comment:
You're missing a full-stop at the end of this sentence.

What makes us think that this might change in the future? (I know we're trying to get rid of globals, but this one seems particularly unlikely to go away soon.)

Reply:
You read my mind, I actually thought that we might be able to get rid of this global - without even searching for usages :see_no_evil: 
As you said; this one will probably be hard to get rid of, so I might as well remove that.